app/src/main/java/sdp/moneyrun/Game.java

```diff
@@ -1,79 +1,240 @@
 package sdp.moneyrun;
 
-import android.location.Location;
 
+import android.location.Location;
+import android.util.Log;
 
+import androidx.annotation.NonNull;
 
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.GenericTypeIndicator;
+import com.google.firebase.database.ValueEventListener;
 
-import java.util.Arrays;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
-
+import java.util.NoSuchElementException;
+import java.util.Objects;
 
 // The entirety of the game logic should be implemented in this class
 public class Game {
-    private final String gameId;
-    private final String name;
-    private List<Player> players;
-    private int maxPlayerCount;
-    private List<Riddle> riddles;
-    private Location startLocation; //TODO: check if we will use the existing or create a new class Location
-    private List<Coin> coins;
-
-    public Game(String gameId, String name, List<Player> players, int maxPlayerCount, List<Riddle> riddles, List<Coin> coins, Location startLocation) {
-        if (players == null || riddles == null || startLocation == null)
+    //Attributes
+    private GameDbData gameDbData;
+
+    //Aux variables
+    private DatabaseReference rootReference;
+    private String id;
+    private static final String TAG = Game.class.getSimpleName();
+    private boolean hasBeenAdded;
+
+    /**
+     * Game constructor
+     * @param name game name
+     * @param players List of players
+     * @param maxPlayerNumber MAximum players
+     * @param startLocation game location
+     */
+    public Game(String name, List<Player> players, int maxPlayerNumber, Location startLocation){
+        if(name == null || players == null || startLocation == null) {
             throw new IllegalArgumentException("Null parameter passed as argument in Game constructor");
-        this.gameId = gameId;
-        this.name = name;
-        this.players = players;
-        this.maxPlayerCount = maxPlayerCount;
-        this.riddles = riddles;
-        this.startLocation = startLocation;
-        this.coins = coins;
+        }
+        rootReference = FirebaseDatabase.getInstance().getReference();
+        gameDbData = new GameDbData(name, players, maxPlayerNumber, startLocation);
+        this.hasBeenAdded = false;
+        this.id = "";
     }
 
-    public String getGameId(){
-        return gameId;
+    private Game(GameDbData data) {
```

Comment:
This should be tested in my opinion

Reply:
Even better, I just removed it, it was never used and I see no point in having it right now if nobody needs it